### PR TITLE
[FIX JENKINS-34496] maven-hpi-plugin injecting test for non hpi packged projects

### PR DESCRIPTION
**Pull request by tfennelly**
See [JENKINS-34496](https://issues.jenkins-ci.org/browse/JENKINS-34496)

@jenkinsci/code-reviewers (esp @jglick)

### CODE REVIEW

File: src/main/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojo.java [hunk 1]
```diff
@@ -47,6 +47,11 @@ private static String quote(String s) {
     }
 
     public void execute() throws MojoExecutionException, MojoFailureException {
+        if (!project.getPackaging().equals("hpi")) {
```
Comment (batmat) [inline src/main/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojo.java]: NIT: I prefer always put the static string on the left side to make NPE impossible
`if ( !"hpi".equals(project.getPackaging()) ) {`

**Comment by batmat** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
🐝

**Comment by stephenc** [comment]
👎 Better would be to fix the actual tests so they don't fail the build as otherwise we can have jelly files in jenkins-modules with missing PIs

**Pull request merged**